DEVELOPMENT.md

```diff
@@ -0,0 +1,162 @@
+
+1. **Prepare Dependencies**
+    - First, navigate to the `graphiql` directory to install and build the project. [See](./graphiql/DEVELOPMENT.md) 
+    - Obtain the following build artifacts from the `graphiql` build:
+        - `graphiql.min.css`
+        - `graphiql.min.js`
+    - Copy these files into the `ontotext-graphql-playground-component/src/components/assets` directory.
+
+2. **Additional Assets**
+    - Additionally, you need to obtain and copy the following files into the same directory (`ontotext-graphql-playground-component/src/components/assets`):
+        - `explorer.index.umd.js`
+        - `explorer.style.css`
+        - `react.development.js`
+        - `react-dom.development.js`
+
+3. **Assets Output**
+    - During the build process, these files are copied to:
+      ```
+      dist/ontotext-graphql-playground-component/assets
+      ```
+
+4. **Build the Component**
+    - Navigate to the `ontotext-graphql-playground-component` directory and build the component for production:
+      ```bash
+      npm run build
+      ```
+
+5. **Publish to npm**
+    - Publish the built component to npm:
+      ```bash
+      npm publish
+      ```
+
+6. **Integration**
+    - Once published, use the component in other projects, such as the GraphDB Workbench.
+
+### 5. **Testing**
+    // TODO describe with cypress install
+
+
+### 6. **Integrating the Component into Another Project**
+
+To integrate the **Ontotext GraphQL Playground Component** into another project, follow these steps:
+
+#### 1. **Install the Component via npm**
+
+Run the following command in your project directory to install the component:
+
+```bash
+npm install ontotext-graphql-playground-component
+```
+
+#### 2. **Ensure Asset Files Are Accessible**
+
+The component will attempt to load the required files from the `assets` folder. You must ensure these files are copied to the correct location in your project during the build process.
+
+##### Example: Using Webpack with `CopyPlugin`
+
+Add the following configuration to your Webpack setup to copy the required assets:
+
+```javascript
+const CopyPlugin = require('copy-webpack-plugin');
+
+module.exports = {
+  // ... other Webpack config
+  plugins: [
+    new CopyPlugin({
+      patterns: [
+        {
+          from: 'node_modules/ontotext-graphql-playground-component/dist/ontotext-graphql-playground-component/assets',
+          to: 'assets',
+        },
+      ],
+    }),
+  ],
+};
+```
+
+#### 3. **Usage Example with Angular (Angular 18)**
+// TODO add description
+
+
+## Best Practices
+
+1. **Submodule Management**
+    - Always update the [`graphiql` submodule](https://github.com/Ontotext-AD/graphiql-fork) in sync with the upstream fork.
+    - Test thoroughly after any updates to ensure no breaking changes affect the StencilJS component.
+
+2. **Code Consistency**
+    - Follow the coding standards of the Ontotext organization for both the `graphiql` submodule and the StencilJS component.
```

Comment:
Development in the `graphiql` must comply with the practices and standarts in that project. This is needed to prevent spreading unwanted changes in the codebase hance preventing conflicts during upgrade.